pattern/src/lib.rs

```diff
@@ -75,6 +173,11 @@ macro_rules! netlist_matches {
     };
 }
 
+/// Like [`netlist_matches!`], but calls
+/// [`replace_value`][Design::replace_value] when a match is found.
```

Comment:
Calls how? I would say that it specifically replaces the value being matched on with the value returned by the rule.